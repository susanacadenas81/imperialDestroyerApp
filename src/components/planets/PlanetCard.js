import PropTypes from 'prop-types';

import './planets.css'

export const PlanetCard = ({ planetName, planetPopulation, planetClimate }) => {
  const name = planetName.toLowerCase().replace(' ', '');
  const imagePath = `/assets/planets/${name}.png`;
  const population = planetPopulation === 'unknown' ? 'Unknown population' : 'Population of ' + planetPopulation/1000 + 'k'
  const change2DefaultImage = (event) => {
    event.target.src = 'assets/default.webp';
  }
  return (
    <div className='planet-card'>
        <div className='image-wrapper'>
            <img className='planet-image' src={ imagePath } alt={ imagePath } onError={change2DefaultImage}></img>
        </div>
        <div className='planet-data-wrapper'>
            <b> { planetName } </b>
            <div> { planetClimate } </div>
            <div> { population } </div>
        </div>
    </div>
  )
}

PlanetCard.propTypes = {
  planetName: PropTypes.string.isRequired,
  planetPopulation: PropTypes.string.isRequired,
  planetClimate: PropTypes.string.isRequired
}
