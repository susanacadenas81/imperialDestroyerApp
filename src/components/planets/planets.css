.planet-list-wrapper {
    display: flex;
    width: 100vw;
    flex-wrap: wrap;
}

.planet-card{
    width: 25em;
    height: 20em;
    display: flex;
    flex-direction: column;
    border: 2px solid rgb(95, 95, 85);
    margin: .5em;
    border-radius: 3px;
    background-color: lightgray;
}

.image-wrapper{
    display: flex;
    justify-content: center;
    background-color: black !important;
    border-bottom: 3px solid rgb(95, 95, 85) !important;
}

.planet-image {
    height: 15em;
    width: 14.5em;
}

.planet-screen-wrapper{
    display: flex;
    max-width: 97vw;
    flex-wrap: wrap;
    padding-left: .5em;
}

.title-icon {
    width: .79em !important;
    height: .79em !important;
    margin-right: .2em;
    padding-right: .2em;
    padding-left: .2em;
    color: #333;
}

.planet-data-wrapper {
    margin-left: .4em;
    padding-top: .1em;
}

.section-title {
    text-align: center;
    font-size: 3em;
    color: black;
}

.header-planet-screen {
    padding: .3em;
    margin-left: .4em;
    width: 100%;
    font-size: .7em;
    font-family: 'Starjedi';
    display: flex;
    align-items: center;
}

hr {
    margin-left: 1.5em;
    margin-right: 1.5em;
    size: 1.5em;
}

@media (max-width: 850px) {
    .planet-card {
        height: inherit !important;
        width: inherit !important;
        max-width: 25em;
    }
    hr {
        display: none;
    }
    .planet-screen-wrapper {
        max-width: inherit;
        width: 120vw;
        justify-content: inherit;

    }
    .planet-list-wrapper {
        width: 100%;
        justify-content: center;
        margin-left: 2em;
    }
}