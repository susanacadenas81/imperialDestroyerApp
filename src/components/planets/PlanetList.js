import { PlanetCard } from './PlanetCard';
import { useFetchData } from '../../hooks/useFetchData';
import './planets.css'
import { useState, useEffect } from 'react';
import { Paginated } from '../ui/Paginated';

export const PlanetList = () => {
  const { count, data, loading } =  useFetchData('planets');
  const [paginatedData, setPaginatedData] = useState([]);
  let sortData = [];

  const onChangePage = (paginatedData) => {
    setPaginatedData(paginatedData)
  }

  useEffect(() => {
    if (!loading) localStorage.setItem('planets', JSON.stringify(data));
  }, [loading])
  
  
  return (
    <>
    
      <div className='planet-list-wrapper'>  
          {
              paginatedData.map((planet) => <PlanetCard key={planet.url} 
                  planetName={planet.name} 
                  planetPopulation={planet.population} 
                  planetClimate={planet.climate} 
              />)
          }
      </div>

      <Paginated 
        onChangePage={ onChangePage } 
        data={ data } 
        sortBy=''
        count={ count }
        loading={ loading }
      />
    </>
  )
}
