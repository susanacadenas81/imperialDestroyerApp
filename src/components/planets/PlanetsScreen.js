import { FaGlobe } from 'react-icons/fa'

import { PlanetList } from "./PlanetList"
import './planets.css'

export const PlanetsScreen = () => {
  return (
    <div className="planet-screen-wrapper">
      <div className="header-planet-screen">
        <h1 className="section-title"><i><FaGlobe className='title-icon' /></i>Planets</h1><hr color='grey' width='77%' />
      </div>
      
      <PlanetList />
    </div>
  )
}
