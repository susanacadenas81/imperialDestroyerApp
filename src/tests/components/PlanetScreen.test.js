import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import { PlanetsScreen } from '../../components/planets/PlanetsScreen';

describe('Tests StarShipsScreen', () => {
    
    const wrapper = shallow(<PlanetsScreen />)

    test('should display correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    })
})