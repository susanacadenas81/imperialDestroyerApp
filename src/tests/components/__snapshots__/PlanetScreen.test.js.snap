// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests StarShipsScreen should display correctly 1`] = `
<div
  className="planet-screen-wrapper"
>
  <div
    className="header-planet-screen"
  >
    <h1
      className="section-title"
    >
      <i>
        <FaGlobe
          className="title-icon"
        />
      </i>
      Planets
    </h1>
  </div>
  <PlanetList />
</div>
`;
